app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,93 @@
+require 'pry'
+class TweetsController < ApplicationController
+    
+   
+    get '/tweets' do
+        if !logged_in?
+          redirect '/login'
+      else 
+        @tweets = Tweet.all 
+        erb :'tweets/index'
+        end
+    end 
+    
+    
+    get '/tweets/new' do
+        if logged_in?
+        erb :'tweets/new' #show new tweets view
+    else 
+        redirect '/login'
+        end
+    end 
+    
+    
+    post '/tweets' do
+     if logged_in?
+        if params[:tweet][:content] != ""
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.